### PR TITLE
VPN-4711 Conda / Android. 

### PR DESCRIPTION
**Pull request by strseb**
This pr adds support for conda / android. 

It's currently blocked, as we still need the rust/android target's to be available in "conda-forge-land". 

It's a very basic pr, similar to conda-windows i add 2 scripts: 
- One that will install all QT dependencies into  `/<envName>/Qt/<Version>` and setup all variables so that `cmake.sh` will pick that up. 
- One other script that will download android's `sdkmanager` and install all dependicies we list in [android_sdk.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/6923/files#diff-fbfa1889efe4dca6c13a2b4aaec289960eaf89ff8ba04834558a6df9df174902)

Blocked by: 
- [x] https://github.com/mozilla-mobile/mozilla-vpn-client/pull/6915
- [x] https://mozilla-hub.atlassian.net/browse/VPN-4805

### CODE REVIEW

File: docs/dev-setup/android.md [hunk 1]
```diff
@@ -1,63 +1,63 @@
...
-Valid architecture values: `x86`, `x86_64`, `armeabi-v7a` `arm64-v8a`, by default it will use all.
 
 # Run
 
 The new apk will be located in
 
     `.tmp/src/android-build/build/outputs/apk/debug/android-build-debug.apk`
+Make sure to Choose the apk-architecture of your env, all others apk's will not work. 
 
 Install with adb on device/emulator
```
Comment (Gela) [inline docs/dev-setup/android.md:62]: Should there be a mention of installing adb in terminal first?

File: docs/dev-setup/android.md [hunk 2]
```diff
@@ -1,63 +1,63 @@
...
```
Comment (Gela) [inline docs/dev-setup/android.md:63]: on line 64

should 
`adb install .tmp/src/android-build/build/outputs/apk/debug/android-build-debug.apk`
be changed to
`adb install .tmp/src/android-build/build/outputs/apk/debug/<ANDROID_BUILD_ARCHITECTURE>.apk`

to emphasize that it needs to be replaced with whatever you build for?

Comment (Gela) [reply docs/dev-setup/android.md:63]: Running `adb install .tmp/.../android.apk` fails with this error: `adb: no devices/emulators found` 

Should there be a step to show how to create an emulator?

File: docs/dev-setup/android.md [hunk 3]
```diff
@@ -1,63 +1,63 @@
...
+$ conda deactivate
+$ conda activate vpn-android
+```
+
+### Setup Android-SDK/NDK and QT. 
+Now install the Android-SDK dependencies and QT via:
+```bash 
+# Setup the Android SDK and NDK for the current Conda env.
+$ ./scripts/android/conda_setup_sdk.sh
+# Setup the Conda env to fetch and Use QT
```
Comment (Gela) [inline docs/dev-setup/android.md:39]:
```suggestion
# Setup the Conda env to fetch and use QT
```

File: env.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 name: vpn
 channels:
-  - conda-forge
+  - conda-forge
```
Comment (Gela) [inline env.yml:3]:
```suggestion
  - conda-forge
```

File: scripts/android/conda_setup_qt.sh [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# QT Android Tools
+python -m aqt install-qt --outputdir $QT_DIR $HOST android ${QT_VERSION} ${ANDROID_ARCH} -m all 
+
+# 
+conda env config vars set QT_HOST_PATH=$QT_DIR/$QT_VERSION/$HOST_FOLDER_NAME
+conda env config vars set QTPATH=$QT_DIR/$QT_VERSION/$ANDROID_ARCH
+
+echo "$QT_DIR/$QT_VERSION/$ANDROID_ARCH/bin/qt-cmake"
+
+echo "finished setup $QT_VERSION QT to Compile for $ANDROID_ARCH"
```
Comment (Gela) [inline scripts/android/conda_setup_qt.sh:40]:
```suggestion
echo "finished setup $QT_VERSION QT to compile for $ANDROID_ARCH"
```

**Review by Gela — CHANGES_REQUESTED** [review]

File: docs/dev-setup/android.md (hunk 1 above)
Comment (strseb) [reply docs/dev-setup/android.md:62]: Good point! - It is already installed :D - Symlinking it to conda `/bin` so you can auto use it

**Review by Gela — APPROVED** [review]

**Pull request merged**